dotcom-rendering/cypress/e2e/parallel-1/sign-in-gate.cy.js

```diff
@@ -188,5 +195,234 @@ describe('Sign In Gate Tests', function () {
 
 			cy.contains('privacy settings');
 		});
+
+		describe('Sign in gate should personalise based on the GU_CO_COMPLETE cookie', function () {
+			it('should show the main sign in gate if GU_CO_COMPLETE if not present', function () {
+				visitArticleAndScrollToGateForLazyLoad();
+				cy.get('[data-cy=sign-in-gate-main]').should('be.visible');
+				cy.get('[data-cy=sign-in-gate-main]').contains(
+					'You need to register to keep reading',
+				);
+				cy.get('[data-cy=sign-in-gate-main]').contains(
+					'It’s still free to read - this is not a paywall',
+				);
+				cy.get('[data-cy=sign-in-gate-main]').contains(
+					'We’re committed to keeping our quality reporting open.',
+				);
+				cy.get('[data-cy=sign-in-gate-main_register]').contains(
+					'Register for free',
+				);
+			});
+
+			// TODO test the flag??
```

Comment:
Left some thoughts on this here: https://github.com/guardian/dotcom-rendering/pull/7112/files#r1098556393